index.bs

```diff
@@ -7,14 +7,110 @@ Group: browser-testing-tools
 URL: https://w3c.github.io/webdriver-bidi/
 No Editor: true
 Abstract: This document defines the BiDirectional WebDriver Protocol, a mechanism for remote control of user agents.
+Default Ref Status: current
+</pre>
+
+<pre class=anchors>
+spec: RFC6455; urlPrefix: https://tools.ietf.org/html/rfc6455
+    type: dfn
+        text: WebSocket ws-URI; url: section-3
+        text: A WebSocket Message Has Been Received; url: section-6.2
+        text: The WebSocket Closing Handshake is Started; url: section-7.1.3
+spec: WEBDRIVER; urlPrefix: https://w3c.github.io/webdriver/
+    type: dfn
+        text: additional capability deserialization algorithm; url: dfn-additional-capability-deserialization-algorithm
+        text: additional WebDriver capability; url: dfn-additional-webdriver-capability
+        text: capability name; url: dfn-capability-name
+        text: error code; url: dfn-error-code
+        text: error; url: dfn-errors
+        text: invalid argument; url: dfn-invalid-argument
+        text: matched capability serialization algorithm; url: dfn-matched-capability-serialization-algorithm
+        text: remote end; url: dfn-remote-ends
+        text: session; url: dfn-sessions
+        text: success; url: dfn-success
 </pre>
 
 Introduction {#intro}
 =====================
 
-[[WEBDRIVER|WebDriver]] defines a protocol for introspection and remote control
-of user agents. This specification extends WebDriver by introducing
-bidirectional communication. In place of the strict command/response
-format of WebDriver, this permits events to stream from the user
-agent to the controlling software, better matching the evented nature
-of the browser DOM.
+[[WEBDRIVER|WebDriver]] defines a protocol for introspection and
+remote control of user agents. This specification extends WebDriver by
+introducing bidirectional communication. In place of the strict
+command/response format of WebDriver, this permits events to stream
+from the user agent to the controlling software, better matching the
+evented nature of the browser DOM.
+
+Transport {#transport}
+======================
+
+Message transport is provided using a [[RFC6455|WebSocket]]
+connection.
+
+A [=remote end=] has a <dfn>connection</dfn>, which is a network connection, and
```

Comment:
Yeah, I guess that makes sense. I'm trying to avoid having too many explicit dependencies on WebDriver, but I suppose if you have multiple sessions in a single instance you'd also have multiple websockets. Although in practice the whole multiple-sessions thing is really about intermediary nodes rather than actual browsers. I don't really know what our feeling about intermediary nodes is in this case.

Reply:
A bit off topic, but aren't multiple sessions pretty normal even without intermediaries? Start multiple instances of the browser to runs tests in parallel, or even have them talk to each other as I know some WebRTC testing setups do.

Reply:
That's still one session per "endpoint node", which basically means "browser instance". Something like ChromeDriver could support multiple sessions of course, but each would be a different browser instance.